ctru-rs/src/services/am.rs

```diff
@@ -0,0 +1,45 @@
+use crate::error::ResultCode;
+use crate::services::fs::FsMediaType;
+
+pub struct Am(());
+
+impl Am {
+    pub fn init() -> crate::Result<Am> {
+        unsafe {
+            ResultCode(ctru_sys::amInit())?;
+            Ok(Am(()))
+        }
+    }
+
+    pub fn get_title_count(&self, mediatype: FsMediaType) -> crate::Result<u32> {
+        unsafe {
+            let mut count = 0;
+            ResultCode(ctru_sys::AM_GetTitleCount(mediatype as u32, &mut count))?;
+            Ok(count)
+        }
+    }
+
+    pub fn get_title_list(&self, mediatype: FsMediaType) -> crate::Result<Vec<u64>> {
+        unsafe {
```

Comment:
as a minor nit, I'd generally recommend keeping the `unsafe` block as small as possible, in this case it can probably just be the calls to `AM_GetTitleList` and `buf.set_len`